### ISSUE TITLE
Feature request: authorization key soft delete

### ISSUE THREAD

**Issue by snobu** [Azure/azure-functions-host#2273]
Currently, when you revoke a host key, it immediately gets deleted. There should be a way to soft-delete a key to emulate proper revocation. An additional remove button (or the same Revoke button that now gets renamed to `Delete`) should take care of cleaning up the key (hard-delete).

![keys](https://user-images.githubusercontent.com/6472374/34482963-2cc7afc8-efc4-11e7-85b8-f17cf009b998.gif)

In configuration, i guess a  boolean `Revoked` property should take care of the logic.

Moreover, a Comment property would also come in handy to know which key gets used where.

Sure, there's API Management that does all that, but it's not a trivial effort to get that going especially when all you need is to keep track of a few API Keys.

**Comment by christopheranderson** [comment]
Makes sense. We can consider this in the future when we're picking up new features.

**Comment by mattchenderson** [comment]
We will not be enhancing the key management APIs in this way. Our recommendation instead would be to use Key Vault for more robust management: https://learn.microsoft.com/azure/azure-functions/function-keys-how-to?tabs=azure-portal#manage-key-storage
